### ISSUE TITLE
'Int32 GetValueOrDefault()' NotSupportedException

### ISSUE THREAD

**Issue by tystol** [hazzik/DelegateDecompiler#78]
I have the exact same expression being used directly within an EF query, and then within a [Computed] property:

```
    db.Timesheets.Select(ts => new 
        {
            TotalHours = ts.TimesheetMode == TimePeriodMode.Duration
                ? ts.DurationHours ?? 0
                : DbFunctions.DiffMinutes(ts.StartTimeOfDay, ts.EndTimeOfDay) / 60m ?? 0m,
        }).ToList();
```

```
   [Computed]
   public decimal TotalHoursDb => TimesheetMode == TimePeriodMode.Duration
        ? DurationHours ?? 0
        : DbFunctions.DiffMinutes(StartTimeOfDay, EndTimeOfDay) / 60m ?? 0m;
```

In the direct version, it works fine. But using DelegateDecompiler, I get the following exception on the computed property:

> An exception of type 'System.NotSupportedException' occurred in EntityFramework.SqlServer.dll but was not handled in user code
> 
> Additional information: LINQ to Entities does not recognize the method 'Int32 GetValueOrDefault()' method, and this method cannot be translated into a store expression.

Am I doing anything wrong?

Or is this a bug? I'm guessing there is some difference between how EF inteprets the above direct expression, and what DelegateDecompiler generates from the IL?

**Comment by hazzik** [comment]
@tystol can you define the property types involved? Or, better just show the full class where this property is located (you can omit properties that are not used in the query)

**Comment by tystol** [comment]
Oops, sorry, of course you'd need that info:

```
public class Timesheet
{
    public TimePeriodMode TimesheetMode { get; set; }
    public TimeSpan? StartTimeOfDay { get; set; }
    public TimeSpan? EndTimeOfDay { get; set; }
    public Decimal? DurationHours { get; set; }
}

public enum TimePeriodMode
{
    Duration = 1,
    StartEndTimes = 2
}
```

**Comment by hazzik** [comment]
@tystol to be honest, I just can not decompile this expression.

**Comment by hazzik** [comment]
@tystol what version of DelegateDecompiler do you use?

**Comment by tystol** [comment]
I might give it a crack a bit later - I haven't looked at the code yet, but any tips? Where's a good starting point?

Using 0.18.1

**Comment by krzaku12** [comment]
Same issue here, using version 0.18.1 of DelegateDecompiler. My property:
`[NotMapped]
[Computed]
public double? IskPerLp
    =>
        (quantity * invTypes.marketData.buyMax - iskCost -
        (lpOfferRequirementsDenormalized.Sum(x => x.quantity * x.invTypes.marketData.sellMin) ?? 0)) /
        (akCost > 0 ? akCost : (lpCost > 0 ? lpCost : 1));`

**Comment by hazzik** [comment]
Released as 0.23.0
